core/src/main/java/utilcalc/core/reportGen/ReportGenUtil.java

```diff
@@ -27,6 +30,23 @@ static List<ServiceCost> splitServiceCostsByReport(
         return result;
     }
 
+    static Map<YearMonth, ServiceCost> mapServiceCostsByMonth(List<ServiceCost> serviceCosts) {
+        return serviceCosts.stream()
+                .flatMap(
+                        cost -> {
+                            LocalDate start = cost.dateRange().startDate();
+                            LocalDate end = cost.dateRange().endDateExclusive().minusDays(1);
+                            List<YearMonth> months = new ArrayList<>();
+                            for (YearMonth ym = YearMonth.from(start);
```

Comment:
Podobny algoritmus uz mame v `DateRange.splitIntoMonthFractions()`. A v `generateHeatingFees` se taky interval prochazi po mesicich, takze to delame trikrat.
Myslim, ze by se cely vypocet heating oddilu dal zjednodusit, kdybychom zustali u `DateRange` a nepouzivali `Map<YearMonth, X>`. Nemam to domyslene a je mozne, ze ta uvaha nikam nepovede, takze navrhuju vyresit ostatni pripominky (jsou to jen drobnosti), zamergovat tohle PR a zkusit refaktorovat v dalsim PR (s tim, ze uz budeme moct vyuzit hotove testy).